src/main/java/org/folio/invoices/utils/HelperUtils.java

```diff
@@ -284,20 +286,16 @@ public static double calculateVoucherAmount(Voucher voucher, List<VoucherLine> v
   public static Double calculateVoucherLineAmount(List<FundDistribution> fundDistributions, List<InvoiceLine> invoiceLines, Voucher voucher) {
     CurrencyUnit invoiceCurrency = Monetary.getCurrency(voucher.getInvoiceCurrency());
     CurrencyUnit systemCurrency = Monetary.getCurrency(voucher.getSystemCurrency());
+    CurrencyConversion conversion = MonetaryConversions.getConversion(systemCurrency, IMF_EXCHANGE_RATE_PROVIDER);
```

Comment:
Sorry to be a pain... I don't think I was clear earlier.  We probably want to avoid "getting" the conversion for each line for two reasons:

1) You could potentially wind up with different conversion rates across your voucher lines if the rate happens to update in the middle of voucher generation (unlikely, but I think possible)
2) Performance.

In `InvoiceHelper.setExchangeRateFactor(...)` you could get the conversion like you do here, get the exchange rate from it and save it so you can later pass it in to this method.  That way you know that the voucher and all the voucher lines have the same converstion/exchangeRate, and you're only incurring the cost of looking up the exchange rate from the provider once.